### PR TITLE
[MOIA-5273] Document offer process

### PR DESCRIPTION
**Pull request by JannikArndt**
Prepared docs for publication:

* restructured to fit flow: Intro > Env > Media Types > Versioning > Errors, then /auth, /customer, /serviceareas, /service-hours, /trip, /trip-activity, /payment
* condensed a lot of duplicate information
* Updated Errors with examples
* Added "Supported Version" for each endpoint (since auth and customer only support json)
* Added code examples for all (currently stable) endpoints
* Added docs for /trip/offer
* Added as much as I know for other trip-endpoints
* Added preliminary docs for trip-activity

This still contains TODOs that are clearly visible, so we _might_ not want to deploy it to prod straight await…
image

### CODE REVIEW

**Comment by JannikArndt** [comment]
How to review (if you have never run `hugo` before): Check out the repo, run 

```bash 
git clone git@github.com:moia-dev/partner-api.git
...
cd partner-api
make api-docs-build 
...
cd api-docs/
./hugo serve
```

and open `localhost:1313`.

File: api-docs/content/main.md [hunk 1]
```diff
@@ -4,38 +4,287 @@
...
+```
+application/json
+application/vnd.moia+json
+application/vnd.moia.v1+json
+```
+
+All calls to our API require a valid token in the `Moia-Auth` header. There are two kinds of tokens: general `client_credentials`-tokens which allow actions such as
+
+* creating customers,
+* logging in as customers,
```
Comment (fstehle) [inline api-docs/content/main.md]: logging in as customers is basically "Authenticating on behalf of a customer", which does not need a token

File: api-docs/content/main.md [hunk 2]
```diff
@@ -4,38 +4,287 @@
...
+
+and _customer-specific tokens_ which can be used to
+
+* request an offer for a trip,
+* order a trip,
+* cancel a trip,
+* give feedback for a trip,
+* load the last trip.
+
+To receive a token, you need to send a `POST` request to `https://api.moia.io/auth/token` that includes your Partner API credentials as Http Basic Auth and the requested `grant_type` as payload.
```
Comment (fstehle) [inline api-docs/content/main.md]: This should be called HTTP basic access authorization header as below

File: api-docs/content/main.md (hunk 1 above)
Comment (JannikArndt) [reply api-docs/content/main.md]:
```suggestion
```

Oh, right

**Comment by JannikArndt** [comment]
We're currently waiting for a `dev` deployment to merge this branch there.

**Pull request merged**